framework/src/executioners/SecantSolve.C

```diff
@@ -0,0 +1,268 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "SecantSolve.h"
+
+#include "Executioner.h"
+#include "FEProblemBase.h"
+#include "NonlinearSystem.h"
+#include "AllLocalDofIndicesThread.h"
+#include "Console.h"
+
+InputParameters
+SecantSolve::validParams()
+{
+  InputParameters params = FixedPointSolve::validParams();
+
+  return params;
+}
+
+SecantSolve::SecantSolve(Executioner * ex) : FixedPointSolve(ex) { allocateStorage(true); }
+
+void
+SecantSolve::allocateStorage(const bool primary)
+{
+  std::string fxn_m1_name;
+  std::string xn_m1_name;
+  std::string fxn_m2_name;
+  std::string xn_m2_name;
+  const std::vector<PostprocessorName> * transformed_pps;
+  std::vector<std::vector<PostprocessorValue>> * transformed_pps_values;
+  if (primary)
+  {
+    fxn_m1_name = "fxn_m1";
+    xn_m1_name = "xn_m1";
+    fxn_m2_name = "fxn_m2";
+    xn_m2_name = "xn_m2";
+    transformed_pps = &_transformed_pps;
+    transformed_pps_values = &_transformed_pps_values;
+  }
+  else
+  {
+    fxn_m1_name = "secondary_fxn_m1";
+    xn_m1_name = "secondary_xn_m1";
+    fxn_m2_name = "secondary_fxn_m2";
+    xn_m2_name = "secondary_xn_m2";
+    transformed_pps = &_secondary_transformed_pps;
+    transformed_pps_values = &_secondary_transformed_pps_values;
+  }
+
+  // TODO: We would only need to store the solution for the degrees of freedom that
+  // will be transformed, not the entire solution.
+  // Store solution vectors for the two previous points and their evaluation
+  auto tag_id = _problem.addVectorTag(xn_m1_name, Moose::VECTOR_TAG_SOLUTION);
```

Comment:
The usage is correct. You may want to save off this tag ids, so that you can use them in `saveVariableValues` function. In this way, you can also avoid the duplicated hard-coded solution vector names.